spring-cql/src/test/java/org/springframework/cassandra/test/integration/core/cql/generator/CqlIndexSpecificationAssertions.java

```diff
@@ -20,23 +20,28 @@
 
 import org.springframework.cassandra.core.keyspace.IndexDescriptor;
 
-import com.datastax.driver.core.ColumnMetadata.IndexMetadata;
+import com.datastax.driver.core.IndexMetadata;
 import com.datastax.driver.core.Session;
+import com.datastax.driver.core.TableMetadata;
 
 public class CqlIndexSpecificationAssertions {
 
 	public static double DELTA = 1e-6; // delta for comparisons of doubles
 
 	public static void assertIndex(IndexDescriptor expected, String keyspace, Session session) {
-		IndexMetadata imd = session.getCluster().getMetadata().getKeyspace(keyspace.toLowerCase())
-				.getTable(expected.getTableName().toCql()).getColumn(expected.getColumnName().toCql()).getIndex();
+		TableMetadata tableMetadata = session.getCluster().getMetadata().getKeyspace(keyspace.toLowerCase())
+				.getTable(expected.getTableName().toCql());
+	    
+		IndexMetadata imd = tableMetadata.getIndex(expected.getName().toCql());
 
-		assertEquals(expected.getName(), imd.getName());
+		assertEquals(expected.getName(), imd == null ? null : imd.getName());
 	}
 
 	public static void assertNoIndex(IndexDescriptor expected, String keyspace, Session session) {
-		IndexMetadata imd = session.getCluster().getMetadata().getKeyspace(keyspace.toLowerCase())
-				.getTable(expected.getTableName().toCql()).getColumn(expected.getColumnName().toCql()).getIndex();
```

Comment:
This is test code, so this isn't really a problem. It's possible to use Cassandra Driver V3 with Cassandra v2 (tested against 2.2.5).